### PR TITLE
divide posts into semesters and a sidebar

### PR DESCRIPTION
**Pull request by yongabyte**
1. fix by adding custom comparator.
 <del>divide posts into semesters and use a object to wrap around the semesters 
 e.g  {fall 2016:[post1, post2..], spring 2016:[ ], .....}
 pitfall: the order of the semesters can't be sorted, because there are the properties of a object.</de>l

2. fix by add hidden-sm on the sidebar
 <del> when the browser's width reduces to a certain amount (e.g a portrait ipad), the side would get under the posts. I have made use of grid </del>.

3. Angularjs might not have native support for bootstrap scrollspy? 
<del> is there a body tag? how can I edit the body tag? I need it for scrollspy. 
[bootstrap scrollspy](http://v4-alpha.getbootstrap.com/components/scrollspy/) </del> 

 #605, #368

### CODE REVIEW

**Comment by agundy** [comment]
If you want to sort the dates you can get a list of the keys, sort that and then use that as a lookup to order the way you want, for now reverse chronological order seems fine though

**Comment by yongabyte** [comment]
The semesters/keys list  does not the "reverse chronological order". it's pure insertion order. It might never cause any problems, but I just think someone should know.

File: client/app/projects/blog/blog.controller.js [hunk 1]
```diff
@@ -80,12 +80,41 @@ angular.module('observatory3App')
...
+    }
+
+    var sortPosts = function(){
+      $scope.semesters = {};
+      for(var i =$scope.posts.length-1; i >-1 ; --i){
+        var postDate = $scope.posts[i].date;
+        var year = postDate.substring(0,4);
+        var semester = '';
+        if(postDate.charAt(5) > 0 || postDate.charAt(6) > 7)
+          semester = 'Fall';
```
Comment (SaswatB) [inline client/app/projects/blog/blog.controller.js]: braces on if and else

**Comment by kmcnellis** [comment]
This fails
Error: [ngRepeat:dupes] Duplicates in a repeater are not allowed. Use 'track by' expression to specify unique keys. Repeater: sem in listOfsem, Duplicate key: string:Fall 2016, Duplicate value: Fall 2016

**Pull request merged**